### PR TITLE
Rename Stimulating{Backend,Device} to Stimulation{Backend,Device}

### PR DESCRIPTION
**Pull request by jougs**
After consultation with @abigailm, we decided to rename all things stimulating to stimulation before we release.

The rationale for this last-second change goes like this:
> The device records something; it is creating a recording; it is a Recording Device.
> The device stimulates something; it is performing a stimulation; it is a Stimulation Device.
> We can't make the second sentence resolve to anything sensible using 'stimulating' instead of 'stimulation'.

As this is the last thing between us and the release of, I would be very grateful if this could be reviewed timely. Thanks!

### CODE REVIEW

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst [hunk 1]
```diff
@@ -11,26 +11,24 @@ it directly from your PyNEST script. Such an external source could be
...
 Technically, this new feature is implemented through stimulation backends that can be based on NEST has been extended by an architecture to select of Nest has been modified to include a backend for
-stimulating devices. This modification is inspired by the backend for
+stimulation devices. This modification is inspired by the backend for
 recording devices. (:doc:`recording from simulations <recording_simulations>`)
 
-With Nest 3.0, we change the terminology of input device to stimulating device.
-Nest 3.0 supports one stimulating backend, `MPI communication`. This backend is
+With Nest 3.0, we change the terminology of input device to stimulation device.
+Nest 3.0 supports one stimulation backend, `MPI communication`. This backend is
 available only when the Nest is compiled with `MPI` support. This is useful in
```
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst:19]:
```suggestion
available only when the NEST is compiled with MPI support. This is useful in
```

**Review by heplesser — APPROVED** [review]
@jougs Nice, I agree with your and @abigailm's evaluation. I just sent you a PR fixing two overlooked occurrences of  "stimulating". The remaining three uses of "stimulating" in our code base are correct as far as I can see. I approve now, subject to you merging my PR.

**Comment by heplesser** [comment]
I think it is a good idea that the doc team also has a look, @jessica-mitchell.

**Comment by clinssen** [comment]
N.B. NESTML has no mentions of "stimulating" that need replacing. cc @pnbabu

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst [hunk 2]
```diff
@@ -11,27 +11,24 @@ it directly from your PyNEST script. Such an external source could be
...
 
 Technically, this new feature is implemented through stimulation backends that can be based on NEST has been extended by an architecture to select of Nest has been modified to include a backend for
-stimulating devices. This modification is inspired by the backend for
+stimulation devices. This modification is inspired by the backend for
 recording devices. (:doc:`recording from simulations <recording_simulations>`)
 
-With Nest 3.0, we change the terminology of input device to stimulating device.
-Nest 3.0 supports one stimulating backend, `MPI communication`. This backend is
-available only when the Nest is compiled with `MPI` support. This is useful in
+With Nest 3.0, we change the terminology of input device to stimulation device.
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst:17]:
```suggestion
With NEST 3.0, we change the terminology of input device to stimulation device.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst [hunk 3]
```diff
@@ -11,27 +11,24 @@ it directly from your PyNEST script. Such an external source could be
...
+Nest 3.0 supports one stimulation backend, `MPI communication`. This backend is
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst:18]:
```suggestion
NEST 3.0 supports one stimulation backend, `MPI communication`. This backend is
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst [hunk 4]
```diff
@@ -11,27 +11,24 @@ it directly from your PyNEST script. Such an external source could be
...
 
-Added the new parameter `stimulus_source`, which can be used to select the
-right backend for the stimulating device. By default, all stimulating devices
-generate their own stimulating signals. If a backend, such as the `MPI` one
-is selected, then the values for producing the stimulating signals are 
-collected using the backend data input channel.
-Added the parameter `label` for the MPI stimulating backend. This parameter
-allows the user to select the file used for storing configuration details of 
-the MPI connection.
+Added the new parameter `stimulus_source`, which can be used to select
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/stimulation_backends.rst:26]:
```suggestion
We added the new parameter ``stimulus_source``, which can be used to select
```

File: doc/userdoc/guides/stimulate_the_network.rst [hunk 5]
```diff
@@ -23,7 +23,7 @@ stimuli for connected nodes. Such data might be explicit lists of
 values, or parametric values as for instance amplitudes and
 frequencies.
 
-.. _stimulating_backends:
+.. _stimulation_backends:
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/stimulate_the_network.rst:26]:
```suggestion
.. _sec_stimulation_backends:
```

Comment (jessica-mitchell) [reply doc/userdoc/guides/stimulate_the_network.rst:26]: Following the styleguide adding the sec precursor. I couldnt find anywhere where this is referenced, but if I missed something please update the ref as well.

**Review by jessica-mitchell — APPROVED** [review]
Just some minor fixes, approving. thanks @jougs

**Comment by jougs** [comment]
> I think it is a good idea that the doc team also has a look, @jessica-mitchell.

@jessica-mitchell is on vacation, so this won't work until the release this afternoon.

**Comment by jougs** [comment]
Forget what I said!

**Comment by jessica-mitchell** [comment]
> @jessica-mitchell is on vacation, so this won't work until the release this afternoon.

Yes, on vacation now but I can keep an eye out today for anything :)

**Pull request merged**